### PR TITLE
Issue #4805: fix HtmlTagCanBeJavadocTag in TokenTypes

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4805

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/api/TokenTypesTest.java [hunk 1]
```diff
@@ -29,9 +29,8 @@
 public class TokenTypesTest {
     @Test
     public void testGetShortDescription() {
-        assertEquals("short description for EQUAL",
-                "The <code>==</code> (equal) operator.", TokenUtils
-                        .getShortDescription("EQUAL"));
+        assertEquals("short description for EQUAL", "The ",
+                TokenUtils.getShortDescription("EQUAL"));
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/TokenTypesTest.java]: This is not clear ...
as we changed first sentence to be "The {@code ==} (equal) operator." in javadoc.
so `TokenUtils.getShortDescription("EQUAL")` should return `The {@code ==} (equal) operator.`

before fix it was:
```
$ mvn clean compile
....
$ grep "^EQUAL" target/classes/com/puppycrawl/tools/checkstyle/api/tokentypes.properties
EQUAL=The <code>==</code> (equal) operator.
```
after fix:
```
$ mvn clean compile
....
$ grep "^EQUAL" target/classes/com/puppycrawl/tools/checkstyle/api/tokentypes.properties
EQUAL=The 
```

That is not acceptable.

Comment (Nimfadora) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/TokenTypesTest.java]: Fixed

**Comment by Nimfadora** [comment]
@romani fixed

**Comment by romani** [comment]
@Nimfadora ,

there is regression in before/after your change:
image

please make sure that there is no diff or new value is correct.
please put to UT, cases `LAND`, `LCURLY`, `SR_ASSIGN`, `SL`, `BSR`.
I agree that `<code>` tag could skip  escaping of symbols, but that should be consistent. 
Case with `LAND` is a bug.

steps to reproduce:
```
mvn clean compile
cp target/classes/com/puppycrawl/tools/checkstyle/api/tokentypes.properties /var/tmp
./fast-forward-merge.sh Nimfadora issue#4805
mvn clean compile
diff /var/tmp/tokentypes.properties target/classes/com/puppycrawl/tools/checkstyle/api/tokentypes.properties
```

**Comment by Nimfadora** [comment]
@romani I cannot revert changes to LCURLY and RCURLY, as it should be escaped in {@code { } as {@code &#123; }. Without escaping build fails.

**Comment by romani** [comment]
> I cannot revert changes to LCURLY and RCURLY 

make sense, it confirmed to keep such escapes in new version of files.

**Pull request merged**